### PR TITLE
chore: package as Python package

### PR DESCRIPTION
**Pull request by ceholden**
## What I'm changing

This PR is submitted in response to https://github.com/GERSL/Fmask/issues/3 and to upstream code changes we've made to make this installable in a maintainable way. Happy to iterate on any changes; this is what's worked for us so far.

This changes packaging only — no science code touched. The goal is to make this installable via `pip`/`uv` and `pixi` without needing to manually set `PYTHONPATH` or run scripts from specific directories.

The flat `src/` modules move into `src/fmask/` (same filenames), the `main/` scripts move into `src/fmask/cli/` and become proper entry points, and the `sys.path` hacks go away. I also added a `fmask-data` command to handle extracting the ancillary data ZIP into the right place, since that's currently a manual step.

### Organization

Bare imports (`from satellite import Landsat`, etc.) became relative imports. CLI scripts use absolute imports so they work both as entry points and when run directly.

### Imports

The original code used `Path(__file__).parent.parent` to find the project root from flat `src/`. One more `.parent` is needed now that modules live in `src/fmask/`. This keeps `data/` and `model/` at the project root, which matches the SharePoint ZIP layout.

### Pre-trained model pickles

**Pickle compat:** The pickled `.pk`/`.pt` models were saved with bare module names (`lightgbmlib`, etc.) so they break after the reorganization. I added a `MetaPathFinder` in `__init__.py` to alias those old names back to `fmask.lightgbmlib` etc. — no model files touched.

### Ecosystem compatibility

- `np.percentile(..., interpolation=)` was renamed to `method=` in NumPy 2.0. One call in `utils.py`.
- `patchify` was listed as a dependency but is never imported anywhere, so I removed it.

### Ancillary data

I added a CLI utility to install the ancillary data (DEMs/water mask) and pre-trained models into a location expected by the Fmask v5 code.

## How to test

```bash
git clone https://github.com/NASA-IMPACT/Fmask-v5
cd fmask
pixi install
pixi run fmask --help
pixi run fmask-data install /path/to/fmask_5_0_1.zip
```

Check pickle compat after loading data:
```bash
pixi run python -c "
import sys, importlib, fmask
mod = importlib.import_module('lightgbmlib')
assert mod is sys.modules['fmask.lightgbmlib']
print('OK:', mod)
"
```

### CODE REVIEW

File: src/fmask/cli/fmask_single.py [hunk 1]
```diff
@@ -433,7 +429,7 @@ def fmask_upl(path_image, dcloud=0, dshadow=0, dsnow = 0, destination = None, sk
     help="Destination directory for results. If not provided, results are saved in the image directory.",
     default="",
 )
-@click.option("--skip_existing", "-s", type=click.Choice(["yes", "no", "Yes", "No", "YES", "NO"]), help="Skip processing if results already exist (set to 0 to force generation).", default="no")
+@click.option("--skip_existing", type=click.Choice(["yes", "no", "Yes", "No", "YES", "NO"]), help="Skip processing if results already exist (set to 0 to force generation).", default="no")
```
Comment (ceholden) [inline src/fmask/cli/fmask_single.py:432]: The `-s` for "skip existing" was conflicting the `-s` for "dilation shadow", so I removed this one to prevent the conflict. This resolves a `click` library warning

File: src/fmask/cli/fmask_batch.py [hunk 2]
```diff
@@ -54,13 +51,13 @@
...
     help="Directory containing Landsat/Sentinel-2 images. Supports multiple images.",
-    default="/Users/shi/Downloads/Validation_Random10",
+    default=".",
 )
 @click.option(
     "--output", "-o",
     type=str,
     help="Destination directory for results. If not provided, results are saved in the resource directory.",
-    default="/Users/shi/Downloads/Validation_Random10_Results_Test",
+    default="",
```
Comment (ceholden) [inline src/fmask/cli/fmask_batch.py:60]: Removes author specific defaults with more generic ones (current directory / resource directory)

File: src/fmask/utils.py [hunk 3]
```diff
@@ -577,7 +578,7 @@ def imfill(data, obsmask, fill_value=None):
     # del obsmask_eroded
 
     # fill the data
-    # a “seed” image, which specifies the values that spread, and a "data" (or "mask") image, which gives the maximum allowed value at each pixel
+    # a "seed" image, which specifies the values that spread, and a "data" (or "mask") image, which gives the maximum allowed value at each pixel
```
Comment (ceholden) [inline src/fmask/utils.py:581]: Removes unicode fancy quotes for normal quotes

File: .gitattributes
Comment (ceholden) [inline .gitattributes:1]: This is an attempt to prevent Git from normalizing the end of line characters that differ between *nix (linux/mac use only the "line feed", LF) and Windows (carriage return and line feed, CRLF). Without it I would see a `git diff` that was entirely the removal of the `^M` carriage return used by Windows

File: .gitignore
Comment (ceholden) [inline .gitignore:1]: Github has a great set of default `.gitignore` they offer when creating a repository, or you can source from their repository,
https://github.com/github/gitignore

This one is the one for Python

File: pixi.toml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+[dependencies]
+python = ">=3.10,<3.13"
+gdal = "*"
+libgdal-jp2openjpeg = "*"
+rasterio = "*"
+numpy = "*"
+scipy = "*"
+pandas = "*"
+scikit-image = "*"
+scikit-learn = "*"
```
Comment (ceholden) [inline pixi.toml:18]: I decided against pinning `scikit-learn` since the pre-trained models are still compatible in the latest version, but pinning to the version used for training is advisable ([ref](https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations)).

We could also explore converting the pre-trained model file into a more portable format like ONNX that won't break if we change versions of `scikit-learn`. See,

- https://scikit-learn.org/stable/model_persistence.html#onnx
- https://onnx.ai/sklearn-onnx/auto_examples/plot_pipeline_lightgbm.html

File: README.md [hunk 5]
```diff
@@ -199,16 +263,16 @@ The global validation samples are available at [this link](https://uconn-my.shar
 - Adapted cloud shadow detection from MATLAB Fmask 4.6 with minor improvements described on [this page](https://github.com/qsly09/fmask5/wiki/Cloud-Shadow-Detection).
 
 #### 1 - 4.6
-Earlier versions of the Fmask tools offered only a physical-rule-based cloud detection module, programmed in MATLAB. See [this page](https://github.com/GERSL/Fmask) for more details.
+Earlier versions of the Fmask tools offered only a physical-rule-based cloud detection module, programmed in MATLAB. See [this page](https://github.com/GERSL/Fmask4) for more details.
```
Comment (ceholden) [inline README.md:266]: Fixed link since the MATLAB based Fmask v4 repository was renamed

**Comment by ceholden** [comment]
Hey @qsly09! I think this PR is ready if you want to take a look and test out the changes. I think I've reconciled all of the changes between the new code you've pushed for v5.0.2 and the code I branched off of (v5.0.1). I'd be particularly curious when you review this to know,

1. Do the code packaging organizational changes still work for you in your workflows (i.e., your HPC environment)
2. Do the changes make sense to you? Feel free to leave questions as PR comments, or we can meet again over video call and talk through them

This should just be a refactor (i.e., no logic or performance changes), but I don't want to propose changes that you wouldn't be comfortable using when making future modifications.

Best,
Chris

**Comment by ceholden** [comment]
Hey @qsly09 - as discussed our new target will be the `v5.0.1` release and **not** the `v5.0.2`.

I would need to drop the commits that made this compatible with `v5.0.2` in order to be mergeable with `v5.0.1`. I could do that, but I would need `main` (or another branch) in your repository to target.

I moved this into a "draft" state to reflect that it's not ready, but it's more likely that this won't be adopted and so I'm just going to close out the PR. For our team we're considering this tagged release as the `v5.0.1` we're adopting,
https://github.com/NASA-IMPACT/Fmask-v5/tree/v5.0.1-1

Here is the comparison against the `v5.0.1` in this repository for what it's worth,
https://github.com/GERSL/Fmask/compare/v5.0.1...NASA-IMPACT:v5.0.1-1
